sdk/python/kfp/components/component_factory.py

```diff
@@ -202,6 +202,8 @@ def extract_component_interface(
         if passing_style in [
                 type_annotations.OutputAnnotation, type_annotations.OutputPath
         ]:
+            if io_name == "Output":
+                raise ValueError('"Output" is an invalid parameter name')
```

Comment:
In both of these lines we should probably reference the variable that sets this `single_output_name_const` instead of repeating the string.